### PR TITLE
Making code easier to read

### PR DESCRIPTION
**Pull request by dvisockas**
Been reading source for the last few hours so thought I'd make it easier to read. 
:heart:

### CODE REVIEW

**Comment by mkorfmann** [comment]
This should be an instamerge!

**Comment by igrigorik** [comment]
I'm ok to merge this, but not sure that spaces are making it that much easier... I'll be the first to admit that the code is a bit of a mess and could really use some love! Ideally, a rubocop-passing reformat -- that would be awesome. Any chance you'd be up for that? :-)

**Comment by mkorfmann** [comment]
+1 for rubocop, thats an awesome idea.

2015-06-26 16:19 GMT+02:00 Ilya Grigorik <notifications@github.com>:

**Comment by dvisockas** [comment]
I will be forking it anyway, so the answer is yes :)

**Comment by dvisockas** [comment]
So I've ran through id3_tree.rb file and tried to fix as much as possible.
Though I didn't fix "Cyclomatic complexity for predict is too high", "Method too long", "Line too long" and few other offences, since they require more time.

Should I just keep commiting on this PR and when I'm done you'll just merge it? :)

**Comment by igrigorik** [comment]
> Should I just keep commiting on this PR and when I'm done you'll just merge it? :)

Yup! And thanks for doing this :)

**Comment by dvisockas** [comment]
I guess I'm done :)

File: lib/decisiontree/id3_tree.rb [hunk 1]
```diff
@@ -161,43 +157,47 @@ def build_rules(tree=@tree)
...
+        if attr[1][key].is_a?(Hash)
           child = attr[1][key].to_a.first[0]
           child_text = "#{child.attribute}\n(#{child.object_id})"
         else
           child = attr[1][key]
           child_text = "#{child}\n(#{child.to_s.clone.object_id})"
         end
-        label_text = "#{key} #{type(attr[0].attribute) == :continuous ? attr[0].threshold : ""}"
+        label_text = "#{key} ''"
+        if type(attr[0].attribute) == :continuous
```
Comment (igrigorik) [inline lib/decisiontree/id3_tree.rb]: this logic breaks example/simple.rb

File: examples/discrete-id3.rb [hunk 2]
```diff
@@ -18,17 +28,31 @@
 dec_tree = DecisionTree::ID3Tree.new(attributes, training, 1, :discrete)
```
Comment (igrigorik) [inline examples/discrete-id3.rb]: discrete-id3.rb:28:in `<main>': undefined local variable or method `attributes' for main:Object (NameError)

**Comment by igrigorik** [comment]
The examples appear to be broken :(

**Comment by dvisockas** [comment]
They're working now :fireworks:

**Comment by igrigorik** [comment]
There's still the type conversion bug: 

```
igrigorik { /git/decisiontree/examples  git:(upstream ⚡ dvisockas-master) 2A} > ruby -I../lib simple.rb
Predicted: sick ... True decision: sick
/Users/igrigorik/git/decisiontree/lib/decisiontree/id3_tree.rb:193:in `gsub!': no implicit conversion of Float into String (TypeError)
```

**Comment by dvisockas** [comment]
Fixed :hammer:

**Comment by igrigorik** [comment]
Great stuff, thanks!

**Pull request merged**